packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -851,71 +449,108 @@ const main = async (opts) => {
+        } catch (err) {
+            console.error('Failed to read from stdin:', err.message)
+            process.exit(1)
+        }
+    } else {
+        answers = await prompt(
+            INITIAL_QUESTIONS,
+            presetId ? {general: {presetOrTemplateId: presetId}} : {}
+        )
     }
 
-    // Add the selected preset to the context object.
-    const selectedPreset = PRESETS.find(
-        ({id}) => id === (presetId || context.answers.general.presetId)
-    )
+    // Determine if the selection is a preset or template.
+    isPreset = PRESETS.some(({id}) => id === answers?.general?.presetOrTemplateId)
+
+    // Update the answer with the actual template id.
+    if (isPreset) {
+        const selectedPreset = PRESETS.find(({id}) => id === answers.general.presetOrTemplateId)
 
-    // Add the preset to the context.
-    context.preset = selectedPreset
+        // NOTE: This is a little weird, but we'll set this value to the template id and treat is as such from this point forward..
+        answers.general.presetOrTemplateId = selectedPreset.templateId
 
-    // If using the preset, output the preset name
-    if (presetId) {
-        console.log(`Using preset "${selectedPreset.name}"`)
+        // Expand the preset answers into the answers object.
+        answers = merge(answers, expandObject(selectedPreset.answers))
     }
 
+    // Since we know we have the template id, we can find the template.
+    selectedTemplate = TEMPLATES.find(({id}) => id === answers.general.presetOrTemplateId)
+
+    // Give some feedback to the user.
+    console.log(`Using template "${selectedTemplate.id}"`)
+
+    // Assign the  preset to the context.
+    context.template = selectedTemplate
+    context.answers = answers
+
     if (!OUTPUT_DIR_FLAG_ACTIVE) {
-        outputDir = p.join(process.cwd(), selectedPreset.id)
+        outputDir = p.join(process.cwd(), selectedTemplate.id)
     }
 
-    // Ask preset specific questions and merge into the current context.
-    const {questions = {}, answers = {}} = selectedPreset
-    if (questions) {
-        const projectAnswers = await prompt(questions, answers)
+    // Ask template specific questions and merge into the current context.
+    // NOTE: Only questions that don't have supplied answers will be asked. This is how we get away with simplifying the code.
+    let {questions} = selectedTemplate
 
-        context = merge(context, {
-            answers: expandObject(projectAnswers)
-        })
-    }
+    // Inquirer doesn't support Regex values for the "validate" property. So lets make a function for it.
+    questions = questions.map((question) => {
+        const validator = VALIDATORS.find(({id}) => id === question.validator)
+
+        return {
+            ...question,
+            validate: validator?.regex
+                ? (input) => new RegExp(validator.regex, 'i').test(input) || validator.message
+                : undefined
```

Comment:
Should we catch errors in case the `validator.regex` is invalid?

Reply:
In the future I think there will be a schema validation step that could be used to do something like this. Right now this is all a design time and if there is an error it's because the pwa-kit developer added something wrong to the program json and it's not something that would effect external developers that are using the script.